compiler/src/dotty/tools/dotc/core/TypeErasure.scala

```diff
@@ -654,7 +654,14 @@ class TypeErasure(sourceLanguage: SourceLanguage, semiEraseVCs: Boolean, isConst
               tr1 :: trs1.filterNot(_.isAnyRef)
             case nil => nil
           }
-        val erasedDecls = decls.filteredScope(sym => !sym.isType || sym.isClass)
+        var erasedDecls = decls.filteredScope(sym => !sym.isType || sym.isClass).openForMutations
```

Comment:
```suggestion
        var erasedDecls = decls.filteredScope(sym => !sym.isType || sym.isClass)
```
? (see comment below)